scripts/control_bar.js

```diff
@@ -4,49 +4,41 @@
 // Construct the image sizing controls.
 var makeImageControl = function(imageElement) {
     let imageKey;
+    // percent need not be an integer but is rounded for display and save
+    // it will typically not be an integer after fit height or width or + or -
     let percent;
     const percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: percent, title: texts.zoomPercent});
```

Comment:
Well spotted. I looked into this some more and also wondered why it doesn't ever report an invalid input. This is because it is not in a form which gets submitted. You can make it report invalid input by calling reportValidity() explicitly. I tried this but it only reports when you press enter from the input and even then not always, at least in Firefox.
The max and min values do constrain what happens with arrow keys or scroll.
The way the code is written now does the range check in addition to the constraint on the input and if you leave out the max and min on the number input it works exactly the same. So I have done this.